### DISCUSSION TITLE
High Availability for RabbitMQ Cluster and Message Topology Operators

### DISCUSSION THREAD

**Discussion by veera-chenna** [rabbitmq/rabbitmq-server#10296] [Questions]
Hi All,

We are trying to check if RabbitMQ Kubernetes Cluster operator and Message Topology Operators supports running with multiple replicas for high availability. We tried searching in RabbitMQ docs but couldn't find anything. Can anyone help us on this?

Also we are looking for recommendations on the resource requests and limits (CPU, Memory) for these operator Kubernetes pods.

Thanks,
Veera

**Comment by mkuratczyk** [comment]
Both operators acquire a lease on startup to make sure only 1 is active (strange things may happen if multiple Operator instances managed the same instance). Therefore, if you run the operator deployment with more than 1 replica, you will have 1 active instance and additional stand-by instances (search for `LEASE_DURATION` for details).

Having said that - what exactly do you think this will accomplish? There's a pretty good chance that if the operator pod terminates for some reason, it will restart faster than than another pod will take over the lease. Keep in mind that the Operator doesn't have to be present for RabbitMQ to work - it's only when `RabbitmqCluster` definitions are changed/added/deleted, that the Operator needs to be running for the changes to happen.
